### ISSUE TITLE
Upstream changes to `win_delay_load_hook.cc`

### ISSUE THREAD

**Issue by Julusian** [nwjs/nw.js#7978]
### Issue Type

Before opening an issue, please search and see if it has already been raised.

- [ ] Bug Report
- [x] Feature Request

- [x] Successfully reproduced against the latest version of NW.js?

> Please use our [mailing list](https://groups.google.com/forum/#!forum/nwjs-general) or [Gitter chatroom](https://gitter.im/nwjs/nw.js) to ask questions. The issue tracker is only for bugs and feature requests, in English only. Please note that issues without a repro or code snippet are less likely to be resolved.

### Current/Missing Behavior

Native modules for windows must be built with a modified version of `win_delay_load_hook.cc`

### Expected/Proposed Behavior

The changes should be upstreamed to node-gyp, so that the file doesn't require manual modification. (I couldn't find an issue/pr for this already, so I don't know if it has been attempted or their view on it)

With [node-api](https://nodejs.org/api/n-api.html) providing a stable across versions of nodejs/v8/electron, I expect the same api stability will apply to nwjs. 
From a quick test, I can produce a build of a node-api based module using your modified `win_delay_load_hook.cc`, and then run that in nodejs and electron. I confirmed the hook was being used by confirming that a build with no `win_delay_load_hook.cc` linked in would fail to load.

So if these changes were pushed to upstream node-gyp, then many more node-api modules would work in nwjs with no effort required from the module maintainer or users

### Additional Info

- Operating System:
- NW.js Version:
- Repro Link:
- Code snippet:
- Crash report:

**Comment by frank-dspeed** [comment]
Good point but won''t fix gyp got deprecated internal by NodeJS and Chromium in chromium we switched to GN and NINJA also NodeJS code gets refactored to use cmake directly. 

You can close that.

**Comment by Julusian** [comment]
@frank-dspeed have a source for your claims?

Even if that is true, that doesnt mean that native modules are being pushed away from gyp. Additionally, node-gyp is still being actively maintained and is wildly popular, so even if it is being phased out, it would make sense to do this to make using native modules in nwjs easier

**Comment by frank-dspeed** [comment]
@Julusian you do not get the full picture please trust me i am from the NodeJS Package Maintainance Group:
- https://github.com/nodejs/package-maintenance/blob/main/MEMBERS.md?plain=1#L28

There exists already gyp to GN Transpilers and other tooling as also i need to tell you that Python is dead it will give no 4.0 version i and the Creator of python do agree in that google it if you do not want to trust me. Wait some month and the world will be a diffrent one there is hugh ongoing overall change around the whole ecosystem this effort crosses many thousend of repositorys.
